# Bangkit! Assignment 5
### Created by:
Harrianto Sunaryan  
Suparjo

## Problem Overview
**Bank Telemarketing** is a method of direct marketing in which a salesperson persuades prospective customers to buy products or services over the phone call or through a subsequent face to face. In this project we will focused on the method based on phone calls.

A lot of customers disturbed by this type of method and resulted affect the relationship between the bank and the customer, especially when they got a continuous phone call in an inappropriate situation suddenly. On the other hand, many salespeople find this method isn't useful because it consumes time with a low chance of success.

If we can predict how successfully the telemarketing method on given customers, the bank can saving time, increase the chance of success, and saving costs by reducing the number of the phone call.

## Data Set Information and Source
The data is related with direct marketing campaigns of a Portuguese banking institution with the binary target wether the customers subscribed a term deposit.

This dataset obtained from https://www.kaggle.com/sharanmk/bank-marketing-term-deposit.
with the credits goes to UCI for providing the data set in https://archive.ics.uci.edu/ml/datasets/bank+marketing# from the research of [Moro et al., 2014].

Citation:
[Moro et al., 2014] S. Moro, P. Cortez and P. Rita. A Data-Driven Approach to Predict the Success of Bank Telemarketing. Decision Support Systems, Elsevier, 62:22-31, June 2014


# Exploratory Data Analysis
In this section, we describe the features and the target, along with the pattern or useful information underlying on them.

In [9]:
import pandas as pd

data=pd.read_csv('bank_customer_survey.csv')
print('Numbers of Observation: ',len(data))
data.head()

Numbers of Observation:  45211


age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47          blue  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome  y  
0  unknown    5   may       261         1     -1         0  unknown  0  
1  unknown    5   may       151         1     -1         0  unknown  0  
2  unknown    5   may        76         1     -1         0  unknown  0  
3  unknown    5   may        92         1     -1         0  unknown  0  
4  unknown    5   may       198         1     -1         0  unknown  0

### Features Description
The data set contains 45211 observations and contains:  
16 features and 1 binary target (0 for failed to convice the customer and 1 for success to convice the customer).

**Features**  
age:
Numerical values represent the customer age.

job:
Categorical strings represent the customer occupation.

martial:
Categorical strings represent the customer marital status.

education:
Categorical strings represent the customer education.

default:
Categorical strings represent wether the customer has default credits or not.

balance:
Numerical values represent the customer average yearly balance in euros.

housing:
Categorical strings represent wether the customer has housing loan or not.

loan:
Categorical strings represent wether the customer has personal loan or not.

contact:
Categorical strings represent contact communication type.

day:
Numerical values represent the last contact day of the month (1-31).

month:
Categorical strings represent the last contact month of the year.

duration:
Numerical values represent last contact durations.

campaign:
Numerical values represent number of contacts performed during this campaign for this customer.

pdays:
Numerical values represent number of days passed after last contacted from a previous campaign. (-1 means the customer was not previously contacted)

previous:
Numerical values represent number of contacts performed before this campaign for this customer.

poutcome:
Categorical strings represent wether the previous campaign was success for this customer.



**Target**  
y:
Categorical strings represent wether the customer subsribed a term of deposit (this campaign).





In [36]:
#Checking data duplicated
print('Number of duplicated :',data.duplicated().value_counts(True)[0]-1)

#Checking Null and data type in each columns
data.info()

Number of duplicated : 0.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 17 columns):
age          45211 non-null int64
job          45211 non-null object
marital      45211 non-null object
education    45211 non-null object
default      45211 non-null object
balance      45211 non-null int64
housing      45211 non-null object
loan         45211 non-null object
contact      45211 non-null object
day          45211 non-null int64
month        45211 non-null object
duration     45211 non-null int64
campaign     45211 non-null int64
pdays        45211 non-null int64
previous     45211 non-null int64
poutcome     45211 non-null object
y            45211 non-null int64
dtypes: int64(8), object(9)
memory usage: 5.9+ MB
